core/common/src/files/FileSystem.kt

```diff
@@ -145,6 +145,20 @@ public sealed interface FileSystem {
      * @throws FileNotFoundException if there is no file or directory corresponding to the specified path.
      */
     public fun resolve(path: Path): Path
+
+    /**
+     * Returns a list of paths corresponding to [directory]'s immediate children.
+     *
+     * If path [directory] was an absolute path, a returned list will also contain absolute paths.
+     * If it was a relative path, a returned list will contain relative paths.
+     *
+     * @param directory a directory to list.
+     * @return a list of [directory]'s immediate children.
+     * @throws FileNotFoundException if [directory] does not exist.
+     * @throws IOException if [directory] points to something other than directory.
+     * @throws IOException if there was an underlying error preventing listing [directory] children.
+     */
+    public fun list(directory: Path): List<Path>
```

Comment:
When looking at other ways to list the directory, folks typically avoid aggregating it in a single list, I suspect for multiple reasons:

* `opendir_at` and ability to be consistent when operating with a collection of paths (also, resolve 
* There are concerns w.r.t. performance and footprint: it is, in fact, a bulk operation that cannot be reasonably limited by the caller (and doesn't make them think to do so)
* [Kotlin-specific] "Lazy, iterable, closeable and countable sequence" is a core primitive for the filesystem, and `list` can be expressed in terms of sequence, while the opposite is not true

Reply:
Probably worth considering a core methods returning a sequence and `toList` as a way to materialize it

Reply:
Then we need a closable sequence (which is not a thing, unlike Java's Stream).

Reply:
There are other alternatives, such as `FS.forEachEntry(dir) { e -> }`.

My point was not to propose another solution but to highlight the potential pitfalls and point out that in other places, folks follow a different path [probably for a reason].
It's totally okay if you considered all of them and disregarded

Reply:
It seems to me that ideally there should be something like what is in stdlib, but I'm not sure that this needs to be done right now.
Is it worth adding an API now that we know in advance that we will remove it soon?!

Reply:
>  soon?!

It depends on how you perceive time, I'm afraid.